### PR TITLE
Feature presidecms 1630 fixing fileuploads on using validateforms

### PR DESCRIPTION
**Pull request by synmal**

### CODE REVIEW

File: system/helpers/presideProxies.cfm [hunk 1]
```diff
@@ -100,19 +100,19 @@
 	</cffunction>
 
 	<cffunction name="validateForms" access="public" returntype="any" output="false">
+		<cfargument name="formData" type="struct" required="true">
```
Comment (DominicWatson) [inline system/helpers/presideProxies.cfm]: I think this will be backward compatible breaking. Any code calling validateForms() will break with this change. What is the purpose of this change and reasoning behind it?

**Review by DominicWatson — CHANGES_REQUESTED** [review]

File: system/helpers/presideProxies.cfm (hunk 1 above)
Comment (synmal) [reply system/helpers/presideProxies.cfm]: In case of using multiple forms, file uploads and `validateForms()`, `validateForms()` do not convert file uploads to binary. Compared to `validateForm()`, we are required to pass in `formData` which the file uploads will be converted to binary along the way.

 However, in `validateForms()` the `formData` that it is passing to `FormsService` is declared in `PresideProxies.cfm`. So the processed `formData` stays in `validateForms()` function. One of the workarounds for this issue is to merge the forms and use `validateForm()` instead.

And regarding the change, I just realised that I had set it to `required="true"`. I will change it to `required="false"`. But I want to know your thoughts regarding this approach first.

Comment (DominicWatson) [reply system/helpers/presideProxies.cfm]: Could you try:

```cfm
<cfargument name="formData" type="struct" default="#getController().getRequestContext().getCollectionWithoutSystemVars()#" />
```

If that solves the issue, then that will also be backwards compatible.

Comment (DominicWatson) [reply system/helpers/presideProxies.cfm]: If that doesn't work, try `getCollection()` rather than `getCollectionWithoutSystemVars()` (there may be a difference with the latter doing a StructCopy() so losing reference to shared data referenced later on)

File: system/helpers/presideProxies.cfm [hunk 2]
```diff
@@ -100,7 +100,7 @@
 	</cffunction>
 
 	<cffunction name="validateForms" access="public" returntype="any" output="false">
-		<cfargument name="formData" type="struct" required="true">
+		<cfargument name="formData" type="struct" default="#getController().getRequestContext().getCollection()#" />
```
Comment (synmal) [inline system/helpers/presideProxies.cfm:103]: Seems like `getCollection()` does fix it. Thanks for the suggestion.

**Comment by DominicWatson** [comment]
Nice one thanks @synmal  - apologies for not picking this up sooner. Merging into upcoming 10.11.0 release.
